protocol/src/main/java/org/fido/iot/protocol/To2ClientService.java

```diff
@@ -221,10 +227,80 @@ protected void doNextOpEntry(Composite request, Composite reply) {
   protected void doSetupDevice(Composite request, Composite reply) {
     getStorage().continuing(request, reply);
 
+    Composite body = request.getAsComposite(Const.SM_BODY);
+    Composite message = Composite.fromObject(getCryptoService().decrypt(body, this.ownState));
+
+    byte[] receivedNonce7 = message.getAsBytes(Const.THIRD_KEY);
+    getCryptoService().verifyBytes(receivedNonce7, nonce7);
+
+    Composite ownerKey2 = message.getAsComposite(Const.FOURTH_KEY);
+    Composite oldCreds = getStorage().getDeviceCredentials();
+    Composite newCreds = Composite.newArray()
+
+        .set(Const.DC_PROTVER,
```

Comment:
New creds is being initialized with old creds. So, the GUID and RV info comparison would always match. Should be initialized with values from 'message'.